cid.go

```diff
@@ -254,6 +254,9 @@ func Decode(v string) (Cid, error) {
 
 	_, data, err := mbase.Decode(v)
 	if err != nil {
+		if len(v) == 46 && v[:2] == "qm" { // https://github.com/ipfs/go-ipfs/issues/7792
+			return Undef, fmt.Errorf("%v: This looks like a CIDv0 that has been lowercased. Convert the CIDv0 to CIDv1 in case-insensitive base32 using 'ipfs cid base32 <Qm..>' and try again.", err)
```

Comment:
Seems like an awfully low level to be handing out command specific advice, since this library has wide use, well beyond go-ipfs. This could arise out of use within Filecoin, or a CAR manipulation library, use-cases where `ipfs` isn't even a thing.

Is there a place above the stack from here, closer to the user, where this check could be inserted instead?

In lieu of that, maybe softening it to seem a little less prescriptive?

```
"%v: This may be a CIDv0 that has been lowercased and cannot be decoded. Base32 CIDv1 is recommended where case-insensitivity is required (e.g. with 'ipfs cid base32 <Qm..>')."
```

Reply:
I agree with Rod; the error message here should be reasonably generic, and IPFS could always do the extra check if go-cid returns an error.